modules/registry-operator-configuration-resource-overview.adoc

```diff
@@ -0,0 +1,72 @@
+// Module included in the following assemblies:
+//
+// * assembly/registry
+// * openshift_images/configuring-registry-operator.adoc
+
+
+[id="registry-operator-configuration-resource-overview-{context}"]
+= Image Registry Operator configuration
+
+The image-registry resource offers the following configuration fields.
+
+[cols="3a,8a",options="header"]
+|===
+|Component |Description
+
+|*ManagementState*
+|*Managed:* the operator will update the registry as configuration resources
+are updated.
+*Unmanaged:* the operator will ignore changes to the configuration resources.
+
+|*Removed*
+|The operator will remove the registry instance and tear down any
+storage that the operator provisioned.
+
+|*Logging*
+|Sets `loglevel` of the registry instance.
+
+|*HTTPSecret*
+|Value needed by the registry to secure uploads, generated by default.
+
+|*Proxy*
+|Not currently implemented. Defines the Proxy to be used when calling master API
+and upstream registries.
+
+|*Storage*
+|*Storagetype:* details for configuring registry storage, for example S3 bucket
+coordinates. Normally configured by default.
+
+|*Requests*
+| API Request Limit details. Controls how many parallel requests a given registry
+instance will handle before queuing additional requests.
+
+|*DefaultRoute*
+|Determines whether or not an external route is defined using the default
+hostname. If enabled, the route uses re-encrypt encryption. Defaults to false.
+
+|*Routes*
+|Array of additional routes to create. You provide the hostname and certificate
+for the route.
+
+|*Replicas*
+|Replica count for the registry.
+|====
+
+== Additional configuration resources
+
+In addition to the `image-registry` resource, additional configuration is
+provided to the operator by separate ConfigMap and Secret resources located
+within the `openshift-image-registry` namespace.
+
+The `image-registry-certificates` (ConfigMap) provides additional CAs for
```

Comment:
This can't be changed by a user anymore. They should create config map in the `openshift-config` namespace and use AdditionalTrustedCA in the image.config.openshift.io resource.

Reply:
Thanks, @dmage! I temporarily commented out the original config content, and put in the following:

"You can create a ConfigMap in the `openshift-config` namespace and use `AdditionalTrustedCA` in the `image.config.openshift.io` resource to contact external registries."

Does that make sense? I think it would also be helpful to provide an example.

Reply:
@bmcelvee 

"You can create a ConfigMap in the `openshift-config` namespace and use it name in `AdditionalTrustedCA` in the image.config.openshift.io resource to provide additional certificate authorities that should be trusted when contacting external registries."

Something like this?.. Maybe it should be a link for another document. For the registry the keys in this ConfigMap don't matter. But for runtime and builds, the config map should look like this:

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: my-registry-ca
data:
  registry.example.com: |
    -----BEGIN CERTIFICATE-----
    ...
    -----END CERTIFICATE-----
  registry-with-port.example.com..5000: |
    -----BEGIN CERTIFICATE-----
    ...
    -----END CERTIFICATE-----
```

The key is the host name of a registry with the port for which this CA is to be trusted. Note that if the registry has the port, like `registry-with-port.example.com:5000`, `:` should be replaced with `..`.

@adambkaplan are there any additional requirements for this from builds?

Reply:
@dmage @bmcelvee worth mentioning that the certIificates must be PEM-encoded (which is the norm). I would stick to the format needed by builds and runtimes in any example.